### PR TITLE
Add extensible tests to transformers

### PR DESCRIPTION
**Pull request by riedgar-ms**
Add basic smoke tests to `transformers` which are parameterised with the name of a Hugging Face model. This is to prevent things like #609 reoccurring.

### CODE REVIEW

File: tests/models/test_transformers.py [hunk 1]
```diff
@@ -1,18 +1,59 @@
...
+
+
+TRANSFORMER_MODELS = ["gpt2", "microsoft/phi-2"]
+
+
+@pytest.mark.parametrize("model_name", TRANSFORMER_MODELS)
+def test_transformer_smoke_gen(model_name):
+    my_model = get_model(f"transformers:{model_name}", trust_remote_code=True)
+
+    prompt = dedent(f"""How many sides has a triangle?""")
```
Comment (Harsha-Nori) [inline tests/models/test_transformers.py:39]: can probably strip the `dedent` here

File: tests/models/test_transformers.py [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
+    lm = my_model + prompt + gen(name="answer", max_tokens=2)
+    assert len(lm["answer"]) > 0, f"Output: {lm['answer']}"
+    # Inexact, but at least make sure not too much was produced
+    assert len(lm["answer"]) < 8, f"Output: {lm['answer']}"
```
Comment (Harsha-Nori) [inline tests/models/test_transformers.py:43]: I like the idea but model quality checks like this (when asserting against unconstrained generations) are a bit dangerous -- you'd be surprised how often a model with minor updates on HuggingFace might spit out an answer that e.g. isn't a number. I'd either add a regex constraint to the gen and check that the number produced can be cast to an integer or just check that the model generated something at all

File: tests/models/test_transformers.py [hunk 3]
```diff
@@ -1,18 +1,59 @@
...
+def test_transformer_smoke_select(model_name):
+    my_model = get_model(f"transformers:{model_name}", trust_remote_code=True)
+
+    prompt = dedent(
+        """How many sides has a triangle?
+
+p) 4
+t) 3
+w) 10
+"""
```
Comment (Harsha-Nori) [inline tests/models/test_transformers.py:56]: small nit, but if we're dedenting, we can probably make this fstring look much nicer (or alternatively, just drop the dedent) call

File: tests/models/test_transformers.py [hunk 4]
```diff
@@ -1,18 +1,59 @@
...
+    )
+    lm = my_model + prompt + select(["p", "t", "w"], name="answer")
+    assert lm["answer"] in ["p", "t", "w"]
```
Comment (Harsha-Nori) [inline tests/models/test_transformers.py:59]: glad to see a forced grammar test here!

File: tests/utils.py [hunk 5]
```diff
@@ -30,7 +30,7 @@ def get_openai_model(model_name, caching=False, **kwargs):
 transformers_model_cache = {}
 
 def get_transformers_model(model_name, caching=False, **kwargs):
-    """ Get an OpenAI LLM with model reuse.
+    """Get model from Hugging Face
```
Comment (Harsha-Nori) [inline tests/utils.py:33]: 😄 nice catch

File: tests/models/test_transformers.py [hunk 6]
```diff
@@ -1,18 +1,59 @@
...
     """
-    assert len(str(lm)) > len("Tweak this proverb to apply to model instructions instead.\n\n")
\ No newline at end of file
+    )
+    assert len(str(lm)) > len(
+        "Tweak this proverb to apply to model instructions instead.\n\n"
+    )
+
+
+TRANSFORMER_MODELS = ["gpt2", "microsoft/phi-2"]
```
Comment (Harsha-Nori) [inline tests/models/test_transformers.py:32]: not for this PR, but we may want to explore different config options tied to each model in a dictionary somewhere in the future. e.g. phi-2 needs to be initialized with `trust_remote_code`, and other HF models have their own quirks too. just a note to keep in mind for later, no need to squeeze in as part of this

File: tests/models/test_transformers.py (hunk 2 above)
Comment (riedgar-ms) [reply tests/models/test_transformers.py:43]: Shouldn't `len()` always return an integer?

File: tests/models/test_transformers.py (hunk 6 above)
Comment (riedgar-ms) [reply tests/models/test_transformers.py:32]: Agreed.

File: tests/models/test_transformers.py (hunk 1 above)
Comment (riedgar-ms) [reply tests/models/test_transformers.py]: Done

File: tests/models/test_transformers.py (hunk 3 above)
Comment (riedgar-ms) [reply tests/models/test_transformers.py]: Done

File: tests/models/test_transformers.py (hunk 4 above)
Comment (riedgar-ms) [reply tests/models/test_transformers.py:54]: My background idea is to have a smoke test for each of the ways a generation (be that `gen()`, `select()` etc.) can be invoked.

**Pull request merged**